PyMca5/PyMcaGui/io/TiledDataChannelTable.py

```diff
@@ -0,0 +1,191 @@
+from PyQt5 import QtWidgets
+
+from PyMca5.PyMcaGui import PyMcaQt as qt
+from PyMca5.PyMcaGui.io.SpecFileCntTable import CheckBoxItem
+
+class TiledDataChannelTable(qt.QTableWidget):
+    """
+    Creates the data channel table (second table) inside the QTiledWidget.
+    The selections for the x and y set the axes in the scan window.
+    """
+    sigTiledDataChannelTableSignal = qt.pyqtSignal(object)
+    def __init__(self, parent=None):
+        qt.QTableWidget.__init__(self, parent)
+        self.dataChannelList = []
+        self.xSelection = []
+        self.ySelection = []
+        self.monSelection = []
+        
+    def format_table(self):
+        """Sets the column headers and the size of the columns for the table."""
+        
+        # Column Labels
+        labels = ['Data Channel', 'x', 'y', 'Mon']
+        self.setColumnCount(len(labels))
+        for i in range(len(labels)):
+            item = self.horizontalHeaderItem(i)
+            if item is None:
+                item = qt.QTableWidgetItem(labels[i])
+            item.setText(labels[i])
+            self.setHorizontalHeaderItem(i, item)
+
+    def clear_table(self):
+        """Clears the table if a different scan is selected."""
+        # Clear contents of the table
+        self.setRowCount(0)
+        self.setColumnCount(0)
+
+        # Reset internal state
+        self.dataChannelList = []
+        self.xSelection = []
+        self.ySelection = []
+        self.monSelection = []
+
+        self.format_table()
+
+    def build_table(self, channelList):
+        """
+        Builds the table in QTiledWidget based on the datachannels in selected
+        scan file.
+        """
+        self.dataChannelList = channelList
+        n = len(channelList)
+        self.setRowCount(n)
+        if n > 0:
+            for i in range(n):
+                self._addLine(i, channelList[i])
+
+    def _addLine(self, i, channelLabel):
+        """
+        Adds individual line to Data Channel Table based on the label of the
+        Data Channel in the scan file.
+        """
+        # Data Channel name
+        item = self.item(i, 0)
+        if item is None:
+            item = qt.QTableWidgetItem(channelLabel)
+            item.setTextAlignment(qt.Qt.AlignHCenter | qt.Qt.AlignVCenter)
+            self.setItem(i, 0, item)
+        else:
+            item.setText(channelLabel)
+        
+        # Data Channel name is not selectable
+        item.setFlags(qt.Qt.ItemIsEnabled)
+
+        # Checkboxes
+        for j in range(1, 4):
```

Comment:
Rather than hard-code the number of columns here, perhaps this value can be extracted from a table property; something like this?

```suggestion
        for j in range(1, self.columnCount()):
```